open-craft/openedx-completion-aggregator
# feat: adds xApiTransforms for completion aggregator events

**Description:** Describe in a couple of sentence what this PR adds

Defines xAPI event transformers and tests for the aggregator events using the base event-routing-backends transformers.

Based heavily on https://github.com/openedx/event-routing-backends/pull/377 -- thank you @andrey-canon!

Part of https://github.com/openedx/openedx-aspects/issues/222

**JIRA:** [FAL-3766](https://tasks.opencraft.com/browse/FAL-3766) (private link)

**Dependencies:**

* https://github.com/open-craft/openedx-completion-aggregator/pull/206
* https://github.com/openedx/event-routing-backends/pull/431

**Installation instructions:**

1. Install the https://github.com/openedx/tutor-contrib-aspects plugin
1. Add this repo+PR branch to `OPENEDX_EXTRA_PIP_REQUIREMENTS`.
1. Add the repo+PR branch for https://github.com/openedx/event-routing-backends/pull/431 to  `OPENEDX_EXTRA_PIP_REQUIREMENTS` too -- have to do this otherwise Aspects will install its own version.
1. Create a course with sections, and subsections, units, and blocks.

**Testing instructions:**

1. In the LMS: complete some blocks, units, subsections and sections.
2. In Superset: login as a superuser, and visit the SQL tab.
   You should be able to see `progressed` and `completed` events corresponding to `chapter`, `sequential`, `vertical`, and `unit` objects in the `xapi.xapi_events_all_parsed_mv` table.

**Reviewers:**
- [ ] @bmtcril 

**Merge checklist:**

- [ ] Revert the TEMP commit and update requirements after https://github.com/openedx/event-routing-backends/pull/431 merges.

**Post merge:**
      finished.
